fix docker to use latest supported node version

NanashiTheNameless opened the pull request:
fix docker to use latest supported node version

NanashiTheNameless committed:
fix dicker to use latest supported node version

NanashiTheNameless commented:
also cleans up trailing whitespace

NanashiTheNameless commented:
(please pardon the typo i was typing fast)

cjerrington commented:
Thank @NanashiTheNameless , Could you update the Node version to 22 in your PR to still use the LTS version. That would still get it upgraded, but on a stable release.

NanashiTheNameless commented:
> Thank @NanashiTheNameless , Could you update the Node version to 22 in your PR to still use the LTS version. That would still get it upgraded, but on a stable release.

Good point I'll do that right now!

NanashiTheNameless committed:
update to LTS node as per request in PR

"node:lts-alpine3.21" is used instead of "node:lts-alpine" as an attempt to midigate the risk of breaking changes

NanashiTheNameless commented:
"node:lts-alpine3.21" is used instead of "node:lts-alpine" as an attempt to mitigate the risk of breaking changes.
currently it is Node 22.15.0 LTS

cjerrington commented:
It looks like `node:lts-alpine` might allow for the LTS of node and it to run on alpine without specifying the version of alpine. 

All of these Docker tags pull the same version: `lts-alpine3.21` https://hub.docker.com/_/node/

[`22-alpine`, `22-alpine3.21`, `22.15-alpine`, `22.15-alpine3.21`, `22.15.0-alpine`, `22.15.0-alpine3.21`, `jod-alpine`, `jod-alpine3.21`, `lts-alpine`, `lts-alpine3.21`⁠](https://github.com/nodejs/docker-node/blob/9bb97e784231f01e61df4d22aaa95a110432a1c7/22/alpine3.21/Dockerfile)

NanashiTheNameless commented:
> It looks like `node:lts-alpine` might allow for the LTS of node and it to run on alpine without specifying the version of alpine.
> 
> All of these Docker tags pull the same version: `lts-alpine3.21` https://hub.docker.com/_/node/
> 
> [`22-alpine`, `22-alpine3.21`, `22.15-alpine`, `22.15-alpine3.21`, `22.15.0-alpine`, `22.15.0-alpine3.21`, `jod-alpine`, `jod-alpine3.21`, `lts-alpine`, `lts-alpine3.21`⁠](https://github.com/nodejs/docker-node/blob/9bb97e784231f01e61df4d22aaa95a110432a1c7/22/alpine3.21/Dockerfile)

I might have misunderstood the versioning, is there any image in particular I should change it to?

NanashiTheNameless committed:
Change to use LTS slim to improve size

After testing it appears node:lts-slim is all that is needed

NanashiTheNameless commented:
Upon testing it seems `node:lts-slim` may be a better choice as it is a minimal install that is only what is strictly needed. 

As there is an option to install dependencies using a minimal install makes more sense.

cjerrington commented:
Thanks for the testing and updates. This has been merged.
